webapplication/aoi/management/commands/_notebook.py

```diff
@@ -131,8 +129,8 @@ def execute_notebook(self):
                     request.error = collected_error
                 request.save(update_fields=['error'])
             with transaction.atomic():
-                request_transaction.save(update_fields=("completed", "rolled_back"))
-                request_transaction.user.save(update_fields=("on_hold", "balance"))
+                request_transaction.save(update_fields=("rolled_back",))
+                request_transaction.user.save(update_fields=("on_hold",))
```

Comment:
If attrs['exit_code'] != 0 - model failed to execute, - then set rolled_back=True, completed=True and change user balance accordingly - release money from on hold.
if attrs['exit_code'] == 0 - model succeeded, - then do not "commit" transaction, that is do not set neither completed, nor change user balance. This must be done in the PublisherThread

Reply:
updated